src/Microsoft.ML.Data/DataView/CompositeRowToRowMapper.cs

```diff
@@ -35,12 +37,15 @@ public CompositeRowToRowMapper(Schema inputSchema, IRowToRowMapper[] mappers)
             OutputSchema = Utils.Size(mappers) > 0 ? mappers[mappers.Length - 1].OutputSchema : inputSchema;
         }
 
-        public Func<int, bool> GetDependencies(Func<int, bool> predicate)
+        /// <summary>
+        /// Given a set of columns, return the input columns that are needed to generate those output columns.
+        /// </summary>
+        IEnumerable<Schema.Column> IRowToRowMapper.GetDependencies(IEnumerable<Schema.Column> columnsNeeded)
         {
-            Func<int, bool> toReturn = predicate;
             for (int i = InnerMappers.Length - 1; i >= 0; --i)
-                toReturn = InnerMappers[i].GetDependencies(toReturn);
-            return toReturn;
+                columnsNeeded = columnsNeeded.Union(InnerMappers[i].GetDependencies(columnsNeeded));
```

Comment:
>Union [](start = 46, length = 5)

Is this needed? The old code seems to "forget" each intermediate mapper's predicate and only return the dependencies of the first one.